packages/esm-patient-vitals-app/src/components/delete-vitals-biometrics-modal/delete-vitals-biometrics.modal.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useCallback } from 'react';
+import { showSnackbar } from '@openmrs/esm-framework';
+import { useTranslation } from 'react-i18next';
+import { deleteEncounter, invalidateCachedVitalsAndBiometrics } from '../../common';
+import { ModalHeader, ModalBody, ModalFooter, Button, InlineLoading } from '@carbon/react';
+import styles from './delete-vitals-biometrics.modal.scss';
+
+interface DeleteVitalsAndBiometricsModalProps {
+  patientUuid: string;
+  encounterUuid: string;
+  closeDeleteModal: () => void;
+}
+
+const DeleteVitalsAndBiometricsModal: React.FC<DeleteVitalsAndBiometricsModalProps> = ({
+  encounterUuid,
+  closeDeleteModal,
+}) => {
+  const { t } = useTranslation();
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  const handleDelete = useCallback(async () => {
+    if (!encounterUuid) {
+      showSnackbar({
+        isLowContrast: false,
+        kind: 'error',
+        title: t('errorDeleting', 'Error deleting vitals and biometrics'),
+        subtitle: t('encounterUuidRequired', 'Encounter UUID is required to delete vitals and biometrics'),
+      });
+      return;
+    }
+
+    setIsDeleting(true);
+    deleteEncounter(encounterUuid)
+      .then(() => {
+        invalidateCachedVitalsAndBiometrics();
+        closeDeleteModal();
+        showSnackbar({
+          isLowContrast: true,
+          kind: 'success',
+          title: t('vitalsAndBiometricsDeleted', 'Vitals and biometrics deleted'),
+        });
+      })
+      .catch((error) => {
+        console.error('Error deleting encounter: ', error);
+        showSnackbar({
+          isLowContrast: false,
+          kind: 'error',
+          title: t('errorDeleting', 'Error deleting vitals and biometrics'),
+          subtitle: error?.message,
+        });
+      })
+      .finally(() => setIsDeleting(false));
+  }, [encounterUuid, t, closeDeleteModal]);
+
+  return (
+    <>
+      <ModalHeader
+        className={styles.modalHeader}
```

Comment:
We shouldn't need this now following https://github.com/openmrs/openmrs-esm-patient-chart/pull/2124.